Remnant hostile hails

temtemy opened the pull request:
This PR will add hostile hails for the Remnant, and varies depending on the interactions of the player (uncontacted, ran away from the blood test, was friends to the Remnant). I did my best to make it match with their attributes (they are isolated human scavengers who escaped the Alpha Wars), but my hails might not be good enough. Feel free to improve them if you want.

(ignore the first two commits, I forgot to separate one commit to another branch, contaminating my master branch)

CyberShadow commented on data/remnant.txt:
```diff
@@ -168,3 +168,115 @@ outfitter "Remnant"
+		"desist and surrender "
+		"surrender yourself "
+		"desist and surrender yourself "
+	word
```
comma?

temtemy replied:
Oh, fixing that

KiLEdEnNis commented:
generally nice, but I think it wouldn't look that good unless there were some more hails (as in, seeing the same combinations every 3 hails isn't good)
plus, usually hails are like

word
"Start"
"Begin"
word
" "
word
"Example"
"Sample"
word (can have a space after them if another sentence will follow after)
"."
"!"

while yours have a space usually at the start or end of a piece of a sentence
try to split such spaces into their own words, to make adding hails easier, which matches most current hails

temtemy commented:
@KiLEdEnNis I changed my hails to make it easier to add more. I also added another hail and some combinations. Feel free to make more suggestions to my PR. :) (I am really new in hails stuff, I wish there was a guide in the wiki for that)

KiLEdEnNis commented on data/remnant.txt:
```diff
@@ -211,12 +226,33 @@ phrase "uncontacted remnant hostile hail"
+	word
+		"have no chance"
+		"have little chance"
+	word
```
should be a word " " before this word

KiLEdEnNis commented on data/remnant.txt:
```diff
@@ -234,11 +270,60 @@ phrase "angered remnant hail"
+		" "
+	word
+		"of us"
+		"of the Remnant"	
```
should be another word after this to add an interpunction (word ".")

KiLEdEnNis commented on data/remnant.txt:
```diff
@@ -234,11 +270,60 @@ phrase "angered remnant hail"
+		"We"
+		"The Remnant"
+		"Our ship"
+	word
```
should be a word " " before this one

KiLEdEnNis commented on data/remnant.txt:
```diff
@@ -234,11 +270,60 @@ phrase "angered remnant hail"
+	word
+		"you"
+		"your ship"
+	word
```
another word " " before this 1

KiLEdEnNis commented on data/remnant.txt:
```diff
@@ -234,11 +270,60 @@ phrase "angered remnant hail"
+		"your ship"
+	word
+		"if you will"
+	word
```
word " "

KiLEdEnNis commented on data/remnant.txt:
```diff
@@ -234,11 +270,60 @@ phrase "angered remnant hail"
+		"human"
+		"pure one"
+		"unaltered"
+	word
```
word " "

KiLEdEnNis commented on data/remnant.txt:
```diff
@@ -248,14 +333,25 @@ phrase "contacted remnant hostile hail"
+	word
+		"should be"
+		"must be"
+	word
```
word " "

KiLEdEnNis commented on data/remnant.txt:
```diff
@@ -248,14 +333,25 @@ phrase "contacted remnant hostile hail"
+		"destroyed"
+		"killed"
+		"out of our space"
+		"gone"
```
missing interpunction at the end of this hail
add a word "." (and/or "!")

KiLEdEnNis commented on data/remnant.txt:
```diff
@@ -248,14 +333,25 @@ phrase "contacted remnant hostile hail"
 	word
-		"Human, if you "
+		"Human,"
+		"Brother,"
```
the game never assumes the players' gender, so this one shouldn't exist as it assumes a male

KiLEdEnNis commented on data/remnant.txt:
```diff
@@ -248,14 +333,25 @@ phrase "contacted remnant hostile hail"
+		"Human,"
+		"Brother,"
+		"Pure one,"
+	word
```
word " "

KiLEdEnNis commented on data/remnant.txt:
```diff
@@ -274,11 +370,17 @@ phrase "contacted remnant hostile disabled hail"
+		" "
 	word
 		"human"
 		"brother"
```
also take this out to keep the game gender-neutral

KiLEdEnNis commented on data/remnant.txt:
```diff
@@ -168,3 +168,221 @@ outfitter "Remnant"
+phrase "contacted remnant hostile disabled hail"
+	word
+		"Human"
+		"Brother"
```
initial comment was hidden in a "show outdated", even though it's not, so I'll repeat:
the game never assumes the player gender, while this assumes male, take this word out

KiLEdEnNis commented on data/remnant.txt:
```diff
@@ -168,3 +168,221 @@ outfitter "Remnant"
+		" "
+	word
+		"human"
+		"brother"
```
> the game never assumes the player gender, while this assumes male, take this word out

same here

KiLEdEnNis commented:
some feedback is hidden in "show outdated" things for no reason, check those as well

temtemy commented on data/remnant.txt:
```diff
@@ -274,11 +370,17 @@ phrase "contacted remnant hostile disabled hail"
+	word
+		" "
+	word
+		"have mercy to me,"
```
A strict search on DuckDuckGo (searched for "have mercy to me", with quotation marks) reveals no results. So yes, this should be "have mercy on me".

Amazinite commented:
I personally don't see the need for different types of hostile hails for every different possible scenario with the Remnant thus far. For example, the "angered" hails for the scenario where the player refuses to give blood suggest that the player can redo their choice, which they can't.

KiLEdEnNis commented:
> suggest that the player can redo their choice, which they can't.

blocking the full Remnant stuff based on 1 choice is a bit bad to me though, we could create some "second" chance (where it's explicitly stated that if the player does it again they'll be locked out)
it's not for this PR but I don't see why we shouldn't add it sometime

Amazinite commented:
That's why I mentioned 
>every different possible scenario with the Remnant *thus far*.

The Remnant are very bare bones in their story so far, so even aside from the unneeded complexity of these hails, things could change that make them obsolete.

endless-sky commented on data/remnant.txt:
```diff
@@ -168,3 +168,243 @@ outfitter "Remnant"
+		"on our worlds"
+		"on our planets"
+		"on Ember Waste"
+		"on this region of the galaxy"
```
That should be "in the Ember Waste" and "in this region..."

endless-sky commented on data/remnant.txt:
```diff
@@ -168,3 +168,243 @@ outfitter "Remnant"
+		" "
+	word
+		"will"
+		"is going to"
```
"We is going to..." doesn't work grammatically.

temtemy commented:
This comment seems unrelated to this PR, but can I ask how to rebase my fork's master with this repo's master? To be clear, make it even with this master (no ahead, no behind). Otherwise, I have to delete my fork, which will also delete my remnant branch, and close this PR. Or I can wait until this PR gets merged/closed, which might take weeks, but I am a bit impatient. :) Or maybe there is another way?

MCOfficer commented:
@temtemy in a clone of your fork (i chose master for this example):
```
// add the main repo as upstream
$ git remote add upstream https://github.com/endless-sky/endless-sky.git

// update master from upstream
$ git checkout master
$ git pull upstream master

// push back to your fork
$ git push
```

temtemy commented:
Oh no, this is why I asked, because my fork is ahead of master. I will still try though, it might work.

temtemy commented:
Okay, atleast in comparisons with my fork and upstream are less weird. Is there a way to delete commits (maybe using git)?

MCOfficer commented:
locally, you can soft-reset your branch back to before whatever you want to revert, then re-issue all commits.
If all commits after the one to be removed are disosable, simply hardreset your branch and it'll all be gone.
if nothing else helps, create a new branch that is uptodate with upstream and cherrypick the commits you want to keep - that's a worst-case solution, though.

temtemy commented:
My master branch don't have commits to be kept, so I hard-reset to the state before I committed and forced update the remote repo. It is now even with master :)

Amazinite commented:
Seeing as how #3761 is on the verge of being merged and would conflict with this (as mentioned previously, the angered hails suggest a second chance, but even with #3761's second chance, the only time you would actually see these angered hails is if you already declined that second chance), I'm closing this. I'd suggest speaking with @Zitchas about implementing these in his next PR relating to the Remnant.
